examples/containers/CounterApp.js

```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+import { Injector } from 'redux';
+import Counter from '../components/Counter';
+import * as CounterActions from '../actions/CounterActions';
+
+export default class CounterApp {
+  render() {
+    return (
+      <Injector actions={CounterActions}>
+        {({ atom: { counter }, actions }) =>
```

Comment:
Every `Injector` is now given the root `atom`. This encourages few `Injector`s at the top. For better performance, just implement `shouldComponentUpdate` on the contained components, or even implement your own finer-grained `Injector`.